alma/alma-php-client
# Update the local docker development experience

This PR fixes the local Docker usage with updated `task` tasks (instead of `make` rules).
The previous Dockerfile was not functional at all due to very outdated software versions.

To test:
```bash
task 5.6:tests
task 5.6:shell
task 7.2:tests
task lint  # Should default to latest PHP version, right now 8.3
```

---

We also do a better job at zipping the dist build, with just what is needed inside:
```bash
task build
```

Result is a ZIP containing:
```
├── src
│  ├── Endpoints
│  ├── Entities
│  ├── Exceptions
│  ├── Lib
│  ├── Payloads
│  ├── Client.php
│  ├── ClientContext.php
│  ├── DependenciesError.php
│  ├── PaginatedResults.php
│  ├── ParamsError.php
│  ├── Request.php
│  ├── RequestError.php
│  └── Response.php
├── CHANGELOG.md
├── composer.json
├── LICENSE
└── README.md
```
